### PR TITLE
Union xnec provider

### PR DESCRIPTION
**Pull request by OpenSuze**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkStateManager.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * @author Hugo Schindler {@literal <hugo.schindler at rte-france.com>}
+ */
+class NetworkStateManager {
+    private final Network network;
+    private final VariantManager variantManager;
+
+    public NetworkStateManager(Network network, XnecProvider xnecProvider) {
+        this.network = network;
+        LossesCompensator.addZeroMWLossesLoadsOnBuses(network);
```
Comment (murgeyseb) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkStateManager.java:21]: If no losses compensation is done, this is useless

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/LossesCompensator.java [hunk 2]
```diff
@@ -47,7 +47,7 @@ void run(Network network) {
         compensateLossesOnBranches(network);
     }
 
-    private void addZeroMWLossesLoadsOnBuses(Network network) {
+    static void addZeroMWLossesLoadsOnBuses(Network network) {
```
Comment (murgeyseb) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/LossesCompensator.java:50]: breaks encapsulation and make this class usage strange as it is both a runner object (run member method) and a static utility class (I do not know is run method is still used by the way)

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/VariantManager.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    private final String defaultVariantId;
+    private final List<Contingency> contingencies;
+
+    VariantManager(Network network, XnecProvider xnecProvider) {
+        this.defaultVariantId = network.getVariantManager().getWorkingVariantId();
+        contingencies = xnecProvider.getContingencies(network);
+    }
+
+    void createAVariantPerContingency(Network network) {
+        if (!contingencies.isEmpty()) {
```
Comment (murgeyseb) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/VariantManager.java:29]: Be careful as the memory usage of the variants may be huge with many contingencies

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/XnecProvider.java [hunk 4]
```diff
@@ -6,19 +6,39 @@
  */
 package com.powsybl.flow_decomposition;
 
+import com.powsybl.contingency.ContingenciesProvider;
 import com.powsybl.iidm.network.Branch;
 import com.powsybl.iidm.network.Network;
+import org.checkerframework.checker.nullness.qual.NonNull;
```
Comment (murgeyseb) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/XnecProvider.java:12]: Perhaps prefer javax.annotation.Nonnull ? Seems nothing perfect available for it currently https://stackoverflow.com/questions/4963300/which-notnull-java-annotation-should-i-use

**Review by murgeyseb — CHANGES_REQUESTED** [review]

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkStateManager.java (hunk 1 above)
Comment (OpenSuze) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkStateManager.java]: fixed

**Review by murgeyseb — APPROVED** [review]

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/LossesCompensator.java (hunk 2 above)
Comment (OpenSuze) [reply flow-decomposition/src/main/java/com/powsybl/flow_decomposition/LossesCompensator.java]: fixed

**Pull request merged**